### ISSUE TITLE
A little problem in utility "ShiftSingleEnd.sh"

### ISSUE THREAD

**Issue by Reemann** [wwang-chcn/wwang_bioinfo_tools#3]
In the last line of `ShiftSingleEnd.sh.sh`, I think it is better to change `if($3-fl/4>0)` to `if($3-fl/4>1)`.
https://github.com/wwang-chcn/wwang_bioinfo_tools/blob/3aee3cc65e86c98f163945fd0c2b76e22c212c88/pipeline/utilities/ShiftSingleEnd.sh#L13

Sometimes when a read in bed is close to the beginning of a chromosome and a little bit short, the `_reads_shift.bed` could be looking like this :
<img width="832" alt="wecom-temp-70d91c5015c1576ec80ec85bf844dbd0" src="https://user-images.githubusercontent.com/74175760/130601146-91bc8a0e-a340-42e1-8279-69f0eac34151.png">
The original read in `_reads.bed` is :
<img width="698" alt="wecom-temp-a93b4cb5a941e83db15669c52e7c097e" src="https://user-images.githubusercontent.com/74175760/130601294-4af1c845-3751-4076-8467-0eed3f0fd888.png">
The END of this read should be 0.5 as `predicted fragment length` is 218. However, `printf "%d"` in awk provides round down integers. So the number 0.5 turn to be zero in this situation.

If the input read file have a line like this ` chrM 0 0`, when using genomeCoverageBed, there will be an Error reporting:
```
Error: malformed BED entry at line 15563228. Start was greater than end. Exiting.
```

In my opinion, maybe changing the condition into `if($3-fl/4>1)` instead of `if($3-fl/4>0)` or using `%.0f` instead of `%d` are better choices.

**Comment by wwang-chcn** [comment]
Actually, chrM\t0\t0 is not a valid bed format. I think we need to filter this recode after genomeCoverageBed. Thanks for your suggestion.

**Comment by wwang-chcn** [comment]
Fixed in 5cfc4b4
